Services/Sessions/WebOSWebAppSession.m

```diff
@@ -746,4 +746,71 @@ - (ServiceSubscription *)subscribeMediaInfoWithSuccess:(SuccessBlock)success fai
     return nil;
 }
 
+#pragma mark - Playlist Control
+
+- (id <PlayListControl>) playListControl
+{
+    return self;
+}
+
+- (CapabilityPriorityLevel)playListControlPriority
+{
+    return CapabilityPriorityLevelHigh;
+}
+
+- (void)playNextWithSuccess:(SuccessBlock)success failure:(FailureBlock)failure {
+    int requestIdNumber = [self getNextId];
+    NSString *requestId = [NSString stringWithFormat:@"req%d", requestIdNumber];
+    NSDictionary *mediaCommand = @{
+                                   @"type" : @"playNext",
+                                   @"requestId" : requestId
+                                   };
+    NSDictionary *message = [self messageForMediaCommand:mediaCommand];
+    [self sendCommandForRequestId:requestId withMessage:message success:success failure:failure];
+}
+
+- (void)playPreviousWithSuccess:(SuccessBlock)success failure:(FailureBlock)failure{
+    int requestIdNumber = [self getNextId];
+    NSString *requestId = [NSString stringWithFormat:@"req%d", requestIdNumber];
+    NSDictionary *mediaCommand = @{
+                                   @"type" : @"playPrevious",
+                                   @"requestId" : requestId
+                                   };
+    NSDictionary *message = [self messageForMediaCommand:mediaCommand];
+    [self sendCommandForRequestId:requestId withMessage:message success:success failure:failure];
+}
+
+- (void)jumpToTrackWithIndex:(NSInteger)index success:(SuccessBlock)success failure:(FailureBlock)failure{
+    int requestIdNumber = [self getNextId];
+    NSString *requestId = [NSString stringWithFormat:@"req%d", requestIdNumber];
+    NSDictionary *mediaCommand = @{
+                                   @"type" : @"jumpToTrack",
+                                   @"requestId" : requestId,
+                                   @"index" : [NSNumber numberWithInteger:index]
+                                   };
+    NSDictionary *message = [self messageForMediaCommand:mediaCommand];
+    [self sendCommandForRequestId:requestId withMessage:message success:success failure:failure];
+}
+
+- (NSDictionary *)messageForMediaCommand:(NSDictionary *)mediaCommand {
+    NSDictionary *message = @{
+                              @"contentType" : @"connectsdk.mediaCommand",
+                              @"mediaCommand" : mediaCommand
+                            };
+    return message;
+}
+
+- (void)sendCommandForRequestId:(NSString *)requestId withMessage:(NSDictionary *)message success:(SuccessBlock)success failure:(FailureBlock)failure{
```

Comment:
You can extend this method to get this duplicated code in:
```
    int requestIdNumber = [self getNextId];
    NSString *requestId = [NSString stringWithFormat:@"req%d", requestIdNumber];
    // add a @"requestId" key to mediaCommand dictionary
    NSDictionary *message = [self messageForMediaCommand:mediaCommand];
    [self sendCommandForRequestId:requestId withMessage:message success:success failure:failure];
```

This will remove duplication from the three methods above.